File: BackEnd_Learn/Java/todolist/src/main/java/com/leeseojune/todolist/error/excetion/ErrorCode.java
package com.leeseojune.todolist.error.excetion;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum ErrorCode {

    BAD_REQUEST(400, "BadRequset."),
    NOT_FOUND(404, "NotFound.");

    private final int status;
    private final String message;

}

File: BackEnd_Learn/kotlin/kopring/src/main/kotlin/io/github/leeseojune53/kopring/global/error/exception/ErrorCode.kt
package io.github.leeseojune53.kopring.global.error.exception

import com.fasterxml.jackson.annotation.JsonFormat

@JsonFormat(shape = JsonFormat.Shape.OBJECT)
enum class ErrorCode(
    val status: Int,
    val message: String
) {

    ALREADY_EXIST_NAME(400, "Already exist name."),
    INVALID_PASSWORD(400, "Invalid password."),
    EXPIRED_TOKEN(401, "Expired token."),
    INVALID_TOKEN(401, "Invalid token"),
    USER_NOT_FOUND(404, "User not found."),
    TOKEN_NOT_FOUND(404, "Token not found")

}